### PR TITLE
Mike

### PR DESCRIPTION
**Pull request by mikegreber**
Basic implementation of firestore, DataManager, and base classes

### CODE REVIEW

File: RocketApp/app/google-services.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (trevor-liu) [inline RocketApp/app/google-services.json:1]: I think the TA mentioned maybe we should share this JSON file privately, maybe we can add this in the .gitignore instead? What do you think?

**Review by trevor-liu — COMMENTED** [review]
Nice work on the classes and Firestore. For the "google-services.json", maybe add it in .gitignore or change our repo to private.

File: RocketApp/app/google-services.json (hunk 1 above)
Comment (mikegreber) [reply RocketApp/app/google-services.json:1]: Yeah I think that is probably a good idea, I will add it to the gitignore.

**Comment by mikegreber** [comment]
Alright I've got a decent base set up for everyone to work off of now. If someone can review and approve it we can merge into main and everyone can have something to work off of.

**Review by trevor-liu — APPROVED** [review]
LGTM

**Pull request merged**